### PR TITLE
Gradle dependencies tidied.

### PR DESCRIPTION
**Pull request by tasomaniac**
Butterknife and Otto are updated

### CODE REVIEW

**Comment by HenriqueRocha** [comment]
:+1:

**Comment by tasomaniac** [comment]
I was also trying to remove unnecessary things. I have just pushed another commit which removes analytics dependency. It is already included in play services. It compiles and it should send data to analytics. 
It will be better if someone with access can check. Thank you.

File: app/build.gradle [hunk 1]
```diff
@@ -71,47 +71,55 @@ android {
...
+    compile 'com.android.support:appcompat-v7:21.0.3'
+    compile 'com.android.support:cardview-v7:21.0.3'
+    compile 'com.android.support:mediarouter-v7:21.0.3'
 
+    //Play Services
     compile 'com.google.android.gms:play-services-location:6.5.87'
     compile 'com.google.android.gms:play-services-games:6.5.87'
     compile 'com.google.android.gms:play-services-identity:6.5.87'
     compile 'com.google.android.gms:play-services-plus:6.5.87'
     compile 'com.google.android.gms:play-services-appstate:6.5.87'
```
Comment (stefanhoth) [inline app/build.gradle]: Random comment: Would be nice to use an ext. variable here for the GMS version.

File: app/build.gradle [hunk 2]
```diff
@@ -71,47 +71,55 @@ android {
...
+    compile 'com.squareup:otto:1.3.6'
     compile 'com.squareup.okhttp:okhttp:1.6.0'
     compile 'com.squareup.okhttp:okhttp-urlconnection:1.6.0'
-    compile 'com.nineoldandroids:library:2.4.0'
     compile 'com.squareup.picasso:picasso:2.3.0'
-    compile 'com.jakewharton:butterknife:5.1.0'
-    compile 'com.google.zxing:android-integration:3.1.0'
-    compile 'com.google.zxing:core:3.1.0'
+
+    //Jake Wharton
```
Comment (stefanhoth) [inline app/build.gradle]: Lol, personal section. :+1:

File: app/build.gradle [hunk 3]
```diff
@@ -71,47 +71,55 @@ android {
...
-    compile 'com.android.support:cardview-v7:21.0.3'
-    compile 'com.android.support:mediarouter-v7:21.0.3'
     compile 'com.google.android.apps.dashclock:dashclock-api:2.0.0'
-    compile 'com.squareup:otto:1.3.4'
+    compile 'com.google.zxing:android-integration:3.1.0'
+    compile 'com.google.zxing:core:3.1.0'
+
+    //Square
+    compile 'com.squareup:otto:1.3.6'
     compile 'com.squareup.okhttp:okhttp:1.6.0'
```
Comment (stefanhoth) [inline app/build.gradle]: 2.x has already been released. Major changes in the API?

File: app/build.gradle (hunk 1 above)
Comment (tasomaniac) [inline app/build.gradle]: Is there an easy way to do it?

File: app/build.gradle (hunk 3 above)
Comment (tasomaniac) [inline app/build.gradle]: I think it introduces major changes. Before 2.x, it was used just like HTTPUrlConnection but there were magic underneath. But after 2.x, API changed.

File: app/build.gradle (hunk 2 above)
Comment (tasomaniac) [inline app/build.gradle]: He deserves it :smile:

File: app/build.gradle (hunk 3 above)
Comment (stefanhoth) [inline app/build.gradle]: Ok, let's leave this for now.

File: app/build.gradle (hunk 1 above)
Comment (stefanhoth) [inline app/build.gradle]: Check this http://tools.android.com/tech-docs/new-build-system/tips#TOC-Controlling-Android-properties-of-all-your-modules-from-the-main-project.

**Comment by stefanhoth** [comment]
During testing with deleted preferences I found that I couldn't sign-in with G+ anymore during onboarding. Can someone verify?

**Comment by tasomaniac** [comment]
What is the seen problem? I cannot verify since I don't have the keys.

**Comment by HenriqueRocha** [comment]
I tested OkHttp 2.2.0 and everything works.
I created a PR #204 with the updates.

**Comment by HenriqueRocha** [comment]
@stefanhoth It's also not working for me when I test on the phone instead of the emulator. I'll investigate the problem, hopefully fix it and submit a PR.

**Comment by HenriqueRocha** [comment]
@stefanhoth I think bug was introduced in commit f1ff725 by @mauimauer a long time ago (which is kind of strange). The app is now crashing because it is doing a `getString()` in the `SharedPreferences` and the value that is put there is of type `Chapter`.

> Key org.gdg.frisbee.CHAPTER expected String but value was a org.gdg.frisbee.android.api.model.Chapter.  The default value <null> was returned.
> Attempt to cast generated internal exception:
> java.lang.ClassCastException: org.gdg.frisbee.android.api.model.Chapter cannot be cast to java.lang.String

The app only crashes if you sign in. I will work on a fix for this during this week and create a PR with all the details.

**Comment by HenriqueRocha** [comment]
@stefanhoth BTW, you can safely merge this since the crash has nothing to do with this PR. It's happening in the develop branch.

**Comment by stefanhoth** [comment]
Thanks for the investigation.

**Pull request merged**